### DISCUSSION TITLE
[Feature] Native Scripting support

### DISCUSSION THREAD

**Discussion by leocb** [immich-app/immich#9711] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I'd like the ability to write and run scripts directly in the Immich UI.

Currently, if I want to, I have to write python scripts and interface with the API, while good for a lot of things, for some simple, one time use operations, I feel like a native script environment would be better.

I think having the ability to do this would simplify the life of power users that want to do bulk operations quickly.

Maybe this could leverage the API internally and/or have native support for dedicated scripting languages, like lua.

Here are few use cases I can think on top of my head:

- Generate albums from folder structure of external library
- Automatic Share/Unshare multiple albums at once
- update photos metadata based on album/folder name (In my case I have scans of physical photos, taken at 1999, scanned now, so the photo appears as 2024)

and much more.

Having the ability to schedule the script to run or a webhook would also be very nice.

Of course, this functionality would need to be explicitly enabled and a big warning would show up with "danger!" or something

Maybe even some way that users can search for "approved" scripts, like the community projects in the docs.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by mmomjian** [comment]
In my opinion this is way too much work for very little benefit (maybe more of a downside tbh). I don't see how this is any easier to a power user than just running a script in a terminal window, and I definitely don't think we want to open the can of worms to approve scripts and make it easier for people to break their install.

**Comment by bo0tzz** [comment]
We're definitely not adding a script runner into Immich. You can just run a script somewhere else that talks to the API, it's really not much hassle ;) If it's something you find yourself doing regularly it might be interesting to set up something like Jupyterlab or Windmill to streamline the "throw a random script at stuff" workflow.

Some of the things you're looking to do will be covered by #1673, and others may be covered by #8173 down the road.
